From: Minfei Huang <mnfhuang@gmail.com>
Date: Sat, 19 Dec 2015 10:54:59 +0800
Subject: [PATCH] ceph: Avoid to propagate the invalid page point
Git-commit: 458c4703aea04674f0b5f63b43cf5669a602a110
Patch-mainline: v4.5
References: Fate#318586


The variant pagep will still get the invalid page point, although ceph
fails in function ceph_update_writeable_page.

To fix this issue, Assigne the page to pagep until there is no failure
in function ceph_update_writeable_page.

Signed-off-by: Minfei Huang <mnfhuang@gmail.com>
Signed-off-by: Yan, Zheng <zyan@redhat.com>
Acked-by: NeilBrown <neilb@suse.com>

---
 fs/ceph/addr.c |    1 -
 1 file changed, 1 deletion(-)

--- a/fs/ceph/addr.c
+++ b/fs/ceph/addr.c
@@ -1159,7 +1159,6 @@ static int ceph_write_begin(struct file
 		page = grab_cache_page_write_begin(mapping, index, 0);
 		if (!page)
 			return -ENOMEM;
-		*pagep = page;
 
 		dout("write_begin file %p inode %p page %p %d~%d\n", file,
 		     inode, page, (int)pos, (int)len);
